src/Magnum/Math/Functions.h

```diff
@@ -410,6 +411,33 @@ template<std::size_t size, class T> inline Vector<size, T> ceil(const Vector<siz
     return out;
 }
 
+
+/**
+@brief Computes a Binomial Coefficient
+
+Given n elements computes the number of ways to choose k elements
+without replacement. The number of ways to do this is
+given by
+@f[
+    \begin{pmatrix} n \\ k \end{pmatrix} = \frac{n! \cdot (n-k)!}{k!} = \frac{n \cdot (n-1) \cdots (n - k + 1)}{k \cdots 1}.
+@f]
+*/
+template<class T> inline typename std::enable_if<IsIntegral<T>::value, T>::type binomialCoefficient(T n, T k) {
+    CORRADE_ASSERT(k >= 0, "k must be positive to compute the Binomial Coefficient", T{0});
```

Comment:
Hmm, what about accepting just `UnsignedInt` here (i.e., making it non-templated)? That would also remove a need for this assert. 

When I was googling for nchoosek, i found [this in matlab docs](http://matlab.izmiran.ru/help/techdoc/ref/nchoosek.html):

> This function is only practical for situations where n is less than about 15.

So I assume there isn't much point in accepting 64-bit integers because the results wouldn't fit into the type anyway?

Oh, and with a non-templated version the definition could then probably go into `Functions.cpp` (don't forget to add `MAGNUM_EXPORT` to the declaration then).

Reply:
Hm I wonder how they arrived at 15. If we compute n choose k the biggest temporary result will 
be (n choose n/2) *  n/2. This overflows for
32 bit unsigned integers for n >= 31
64 bit unsigned integers for n >= 63
128 bit unsigned integer for n >= 126
So I do think there is some value in having this as a template. But since I only need n <= 30 I can untemplate it if you still think this is the better way :)

Reply:
Okay, what about `UnsignedInt` inputs and `UnsignedLong` output (and temporaries)? :)

I was thinking about something like "factorials REALLY HARD to compute for n > 15" but that doesn't really seem to make sense either. Maybe in 1994 it was and that's why the 15?

Reply:
ok :).
If they really compute the factorial that would already get problematic for n = 13.
Another possibility is that they had 16 bit unsigned integers that actually gets you to [15](https://www.wolframalpha.com/input/?i=%2815+choose+15%2F2%29+*++15%2F2+%3E+2%5E16+-+1+) if am not mistaken.

Reply:
Yep, that sounds likely, maybe the documentation is really from the time where 32bit calculations were expensive and just hasn't been updated since.